### PR TITLE
test(source): unit-tests overhaul

### PR DESCRIPTION
**Pull request by louis-bompart**
## Proposed changes

While working on the improvement on the `IsAuthenticated` I hit some hurdles on the source plugin UT due to bad scoping (either on 'what' should be tested or 'what' should be mocked).

This fix that and more:
 - Replaced complicated expect statements by  Jest Snapshot expectation (it's pretty lazy, but it works nicely)
 - Actually test `source:list`: We we're still running `source:push:list` in the spec file 🤣 
 - Test the 'you should not delete that many stuff with `source:push:delete`' logic.
 
This should ease future changes in the 'common' stuff & simplify the maintenance of the tests, while also ensuring better coverage. ('cause the jest snapshot will catch more changes)

## Testing

- [x] Unit Tests.

### CODE REVIEW

**Review by y-lakhdar — APPROVED** [review]
It took me quite some time to go over the PR but very nice work on the unit test simplification.

File: packages/cli/source/src/commands/source/catalog/__snapshots__/add.spec.ts.snap [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`source:catalog:add when the batch upload fails returns an information message on add failure from the API 1`] = `
+"Error while trying to add document.
+
+Status code: 412
+Error code: [31mBAD_REQUEST[39m
+Message: [31mthis is a bad request and you should feel bad[39m
+Request ID: [31mundefined[39m"
```
[the remark this answers is not in the document]
Comment (louis-bompart) [reply packages/cli/source/src/commands/source/catalog/__snapshots__/add.spec.ts.snap]: Already resolved

File: packages/cli/source/src/commands/source/push/add.ts [hunk 2]
```diff
@@ -79,7 +79,7 @@ export default class SourcePushAdd extends CLICommand {
 
     startSpinner('Processing files');
 
-    const fileNames = await getFileNames(flags);
+    const fileNames = getFileNames(flags, SourcePushAdd.id);
```
[the remark this answers is not in the document]
Comment (louis-bompart) [reply packages/cli/source/src/commands/source/push/add.ts:82]: From add.spec.ts.snap:
`exports[`source:push:add when the batch upload is successful throws when no flags are specified 1`] = `"You must set the \`files\` flag. Use \`source:push:add --help\` to get more information."`;`

This prove & test that it works ✔️

File: packages/cli/source/src/commands/source/push/new.ts [hunk 3]
```diff
@@ -33,6 +33,7 @@ export default class SourcePushNew extends CLICommand {
     IsAuthenticated(),
     HasNecessaryCoveoPrivileges(writeSourceContentPrivilege)
   )
+  //TODO: Refactor with catalog
```
[the remark this answers is not in the document]
Comment (louis-bompart) [reply packages/cli/source/src/commands/source/push/new.ts:36]: reworded

File: packages/cli/source/src/commands/source/__snapshots__/list.spec.ts.snap [hunk 4]
```diff
@@ -0,0 +1,16 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`source:push:list works when the user has access to a list of push source 1`] = `
+" Id     Name               Owner Source type Source visibility Status     Number of documents 
+ ────── ────────────────── ───── ─────────── ───────────────── ────────── ─────────────────── 
+ the_id the_id_displayName bob               SECURED           PUSH_READY 1234
```
[the remark this answers is not in the document]
Comment (louis-bompart) [reply packages/cli/source/src/commands/source/__snapshots__/list.spec.ts.snap:6]: fixed mock

File: packages/cli/source/src/commands/source/__snapshots__/list.spec.ts.snap [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+ ────── ────────────────── ───── ─────────── ───────────────── ────────── ─────────────────── 
+ the_id the_id_displayName bob               SECURED           PUSH_READY 1234                
+"
+`;
+
+exports[`source:push:list works when there is no push source configured 1`] = `
+"There is no push nor catalog source in organization foo
+You can:
+* create a push source using source:push:new
+* create a catalog source using source:catalog:new
```
[the remark this answers is not in the document]
Comment (louis-bompart) [reply packages/cli/source/src/commands/source/__snapshots__/list.spec.ts.snap:14]: changed

**Pull request merged**